test/ERC20VotesPartialDelegationUpgradeable.t.sol

```diff
@@ -770,6 +770,154 @@ contract Transfer is PartialDelegationTest {
   }
 }
 
+contract Permit is PartialDelegationTest {
+  /**
+   * @dev Indicates a failure with the `spender`’s `allowance`. Used in transfers.
+   * @param spender Address that may be allowed to operate on tokens without being their owner.
+   * @param allowance Amount of tokens a `spender` is allowed to operate with.
+   * @param needed Minimum amount required to perform a transfer.
+   */
+  error ERC20InsufficientAllowance(address spender, uint256 allowance, uint256 needed);
+
+  function testFuzz_TransferWithPermit(
+    uint256 _holderSeed,
+    address _receiver,
+    uint256 _transferAmount,
+    uint256 _existingBalance,
+    uint256 _deadline
+  ) public {
+    _holderSeed = bound(
+      _holderSeed,
+      1,
+      /* private key can't be bigger than secp256k1 curve order */
+      115_792_089_237_316_195_423_570_985_008_687_907_852_837_564_279_074_904_382_605_163_141_518_161_494_337 - 1
+    );
+    address _holder = vm.addr(_holderSeed);
+    _deadline = bound(_deadline, block.timestamp, type(uint256).max);
+    vm.assume(_holder != address(0));
+    vm.assume(_receiver != address(0) && _receiver != _holder);
+    _transferAmount = bound(_transferAmount, 0, type(uint208).max);
+    _existingBalance = bound(_existingBalance, _transferAmount, type(uint208).max);
+
+    vm.startPrank(_holder);
+    tokenProxy.mint(_existingBalance);
+    bytes32 _message = keccak256(
+      abi.encode(
+        keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"),
+        _holder,
+        _receiver,
+        _transferAmount,
+        tokenProxy.nonces(_holder),
+        _deadline
+      )
+    );
+    bytes32 _messageHash = keccak256(abi.encodePacked("\x19\x01", DOMAIN_SEPARATOR, _message));
+    (uint8 _v, bytes32 _r, bytes32 _s) = vm.sign(_holderSeed, _messageHash);
+
+    tokenProxy.permit(_holder, _receiver, _transferAmount, _deadline, _v, _r, _s);
+    vm.stopPrank();
+
+    vm.prank(_receiver);
+    tokenProxy.transferFrom(_holder, _receiver, _transferAmount);
+    assertEq(tokenProxy.balanceOf(_receiver), _transferAmount);
+  }
+
+  function testFuzz_RevertIf_InvalidReceiverTriesToUseAValidSignature(
```

Comment:
IMO this test is testing `transferFrom`, not `permit`. Do you agree? In that case, we can delete (this test and next test too). for sad path `permit` tests, try messing with any of these params: _holder, _receiver, _transferAmount, _deadline, or... _v, _r, _s (you can mess with the message/message hash to get different v,r,s). we want to see the permit call revert.

Reply:
Agreed, testing for the custom errors inside the `permit` function now instead in [82c57fe](https://github.com/voteagora/ERC20VotesPartialDelegationUpgradeable/pull/41/commits/82c57feb58fb4f9463eb45bb4778cf912954003a)